core/new-gui/src/app/workspace/component/workflow-editor/mini-map/mini-map.component.ts

```diff
@@ -1,247 +1,75 @@
 import { auditTime } from "rxjs/operators";
 import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
-
-export const MINI_MAP_WRAPPER_ID = "texera-mini-map-wrapper";
-export const MINI_MAP_JOINTJS_MAP_ID = "texera-mini-map-editor-jointjs-body-id";
-export const MINI_MAP_NAVIGATOR_ID = "mini-map-navigator-id";
-export const MINI_MAP_GRID_SIZE = 45;
-
-/**
- * MiniMapComponent is the componenet that contains the mini-map of the workflow editor component.
- *  This component is used for navigating on the workflow editor paper.
- *
- * The mini map component is bound to a JointJS Paper. The mini map's paper uses the same graph/model
- *  as the main workflow (WorkflowEditorComponent's model), making it so that the map will always have
- *  the same operators and links as the main workflow.
- *
- * @author Cynthia Wang
- * @author Henry Chen
- */
+import { MAIN_CANVAS_LIMIT } from "../workflow-editor-constants";
+import { WORKFLOW_EDITOR_JOINTJS_WRAPPER_ID } from "../workflow-editor.component";
+import * as joint from "jointjs";
 @UntilDestroy()
 @Component({
   selector: "texera-mini-map",
-  templateUrl: "./mini-map.component.html",
-  styleUrls: ["./mini-map.component.scss"],
+  templateUrl: "mini-map.component.html",
+  styleUrls: ["mini-map.component.scss"],
 })
 export class MiniMapComponent implements AfterViewInit {
-  // the DOM element ID of map. It can be used by jQuery and jointJS to find the DOM element
-  // in the HTML template, the div element ID is set using this variable
-  public readonly MINI_MAP_WRAPPER_ID = MINI_MAP_WRAPPER_ID;
-  public readonly MINI_MAP_JOINTJS_MAP_ID = MINI_MAP_JOINTJS_MAP_ID;
-  public readonly MINI_MAP_NAVIGATOR_ID = MINI_MAP_NAVIGATOR_ID;
-  public readonly MINI_MAP_GRID_SIZE = 45;
-
-  public readonly MAIN_PAPER_JOINTJS_WRAPPER_ID = WORKFLOW_EDITOR_JOINTJS_WRAPPER_ID;
-
-  public MINI_MAP_ZOOM_SCALE = 0;
-  public MINI_MAP_SIZE = {
-    width: 0,
-    height: 0,
-  };
-
-  public show: boolean = true;
-
-  private mouseDownPosition: Point | undefined;
-  private miniMapPaper: joint.dia.Paper | undefined;
-
+  scale = 0;
   constructor(private workflowActionService: WorkflowActionService) {}
-
   ngAfterViewInit() {
-    this.initializeMapPaper();
-    this.handleMouseEvents();
-    this.handleWindowResize();
-  }
-
-  public handleMouseEvents() {
-    const navigatorElement = document.getElementById(this.MINI_MAP_NAVIGATOR_ID);
-    if (navigatorElement == null) {
-      throw new Error("minimap: cannot find navigator element");
-    }
-
-    fromEvent<MouseEvent>(navigatorElement, "mousedown")
+    const map = document.getElementById("mini-map")!;
+    this.scale = map.offsetWidth / (MAIN_CANVAS_LIMIT.xMax - MAIN_CANVAS_LIMIT.xMin);
+    new joint.dia.Paper({
+      el: map,
+      model: this.workflowActionService.getJointGraphWrapper().jointGraph,
+      background: { color: "#F6F6F6" },
+      interactive: false,
+      width: map.offsetWidth,
+      height: map.offsetHeight,
+    })
```

Comment:
just wonder if this can be initalized in the constructor, or it has to be initialized after view init? maybe it requires the view to know the current minimap width and height for scale calculation?

Reply:
It has to be done after view initialization; otherwise, the HTML element will be null, and we cannot bind the paper to the HTML element.